pkg/tsdb/influxdb/influxql/converter/converter.go

```diff
@@ -0,0 +1,395 @@
+		if err != nil {
+			return rspErr(err)
+		}
+		switch l1Field {
+		case "results":
+			rsp = readResults(iter, frameName, query)
+			if rsp.Error != nil {
+				return rsp
+			}
+		case "":
+			if err != nil {
+				return rspErr(err)
+			}
+			break l1Fields
+		default:
+			v, err := iter.Read()
+			if err != nil {
+				rsp.Error = err
+				return rsp
+			}
+			fmt.Println(fmt.Printf("[ROOT] TODO, support key: %s / %v\n\n", l1Field, v))
+		}
+	}
+
+	return rsp
+}
+
+func readResults(iter *jsonitere.Iterator, frameName []byte, query *models.Query) *backend.DataResponse {
+	var rsp *backend.DataResponse
+l1Fields:
+	for more, err := iter.ReadArray(); more; more, err = iter.ReadArray() {
+		if err != nil {
+			return rspErr(err)
+		}
+		for l1Field, err := iter.ReadObject(); l1Field != ""; l1Field, err = iter.ReadObject() {
+			if err != nil {
+				return rspErr(err)
+			}
+			switch l1Field {
+			case "series":
+				rsp = readSeries(iter, frameName, query)
+			case "":
+				break l1Fields
+			default:
+				_, err := iter.Read()
+				if err != nil {
+					return rspErr(err)
+				}
+			}
+		}
+	}
+
+	return rsp
+}
+
+func readSeries(iter *jsonitere.Iterator, frameName []byte, query *models.Query) *backend.DataResponse {
+	var measurement string
+	var tags map[string]string
+	var columns []string
+	var valueFields data.Fields
+	var hasTimeColumn bool
+	rsp := &backend.DataResponse{Frames: make(data.Frames, 0)}
+	for more, err := iter.ReadArray(); more; more, err = iter.ReadArray() {
+		if err != nil {
+			return rspErr(err)
+		}
+
+		for l1Field, err := iter.ReadObject(); l1Field != ""; l1Field, err = iter.ReadObject() {
+			if err != nil {
+				return rspErr(err)
+			}
+			switch l1Field {
+			case "name":
+				if measurement, err = iter.ReadString(); err != nil {
+					return rspErr(err)
+				}
+			case "tags":
+				if tags, err = readTags(iter); err != nil {
+					return rspErr(err)
+				}
+			case "columns":
+				columns, err = readColumns(iter)
+				if err != nil {
+					return rspErr(err)
+				}
+				if columns[0] == "time" {
+					hasTimeColumn = true
+				}
+			case "values":
+				valueFields, err = readValues(iter, hasTimeColumn)
+				if err != nil {
+					return rspErr(err)
+				}
+			default:
+				v, err := iter.Read()
+				if err != nil {
+					return rspErr(err)
+				}
+				fmt.Printf("unsupported key: %s / %v\n", l1Field, v)
+			}
+		}
+
+		if util.GetVisType(query.ResultFormat) == util.TableVisType {
+			handleTableFormatFirstFrame(rsp, measurement, query)
+			handleTableFormatFirstField(rsp, valueFields, columns)
+			handleTableFormatTagFields(rsp, valueFields, tags)
+			handleTableFormatValueFields(rsp, valueFields, tags, columns)
+		} else {
+			// time_series response format
+			if hasTimeColumn {
+				// Frame with time column
+				newFrames := handleTimeSeriesFormatWithTimeColumn(valueFields, tags, columns, measurement, frameName, query)
+				rsp.Frames = append(rsp.Frames, newFrames...)
```

Comment:
I would return `rsp` also here and remove the `else` below.

Reply:
Cannot return rsp here either